subsys/debug/asan.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2019 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_SOC_POSIX)
```

Comment:
Why are you switching on SOC_POSIX instead of the POSIX ARCH?

Reply:
If you only hook into the asan library with __asan_default_options then it should be better to use `CONFIG_ARCH_POSIX`.
But if you add a native task, the part where you add the native task would be better be under a `#if defined(CONFIG_SOC_POSIX)`